oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -326,6 +338,48 @@ public void getDefaultCredentials_envUser_providesToken() throws IOException {
     testUserProvidesToken(testProvider, USER_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN);
   }
 
+  @Test
+  public void getDefaultCredentials_envGdchServiceAccount_providesToken() throws IOException {
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    InputStream gdchServiceAccountStream =
+        GdchCredentialsTest.writeGdchServiceAccountStream(
+            GDCH_SA_FORMAT_VERSION,
+            GDCH_SA_PROJECT_ID,
+            GDCH_SA_PRIVATE_KEY_ID,
+            GDCH_SA_PRIVATE_KEY_PKC8,
+            GDCH_SA_SERVICE_IDENTITY_NAME,
+            GDCH_SA_CA_CERT_PATH,
+            GDCH_SA_TOKEN_SERVER_URI);
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    String gdchServiceAccountPath = tempFilePath("gdch_service_account.json");
+    testProvider.addFile(gdchServiceAccountPath, gdchServiceAccountStream);
+    testProvider.setEnv(DefaultCredentialsProvider.CREDENTIAL_ENV_VAR, gdchServiceAccountPath);
+
+    GoogleCredentials defaultCredentials = testProvider.getDefaultCredentials(transportFactory);
+
+    assertNotNull(defaultCredentials);
+    assertTrue(defaultCredentials instanceof GdchCredentials);
+    assertEquals(GDCH_SA_PROJECT_ID, ((GdchCredentials) defaultCredentials).getProjectId());
+    assertEquals(
+        GDCH_SA_SERVICE_IDENTITY_NAME,
+        ((GdchCredentials) defaultCredentials).getServiceIdentityName());
+    assertEquals(
+        GDCH_SA_TOKEN_SERVER_URI, ((GdchCredentials) defaultCredentials).getTokenServerUri());
+    assertNull(((GdchCredentials) defaultCredentials).getApiAudience());
+
+    defaultCredentials =
+        ((GdchCredentials) defaultCredentials).createWithGdchAudience(GDCH_SA_API_AUDIENCE);
+    transportFactory.transport.addGdchServiceAccount(
+        GdchCredentials.getIssSubValue(GDCH_SA_PROJECT_ID, GDCH_SA_SERVICE_IDENTITY_NAME),
+        ACCESS_TOKEN);
+    transportFactory.transport.setTokenServerUri(GDCH_SA_TOKEN_SERVER_URI);
+
+    assertNotNull(defaultCredentials);
```

Comment:
Lets remove this part, we cannot have a full test, but we make sure GDCH credential is recognized and properties are correct

Reply:
done